src/components/Guilds/common/Modal/transaction.tsx

```diff
@@ -0,0 +1,128 @@
+import PendingCircle from 'components/common/PendingCircle';
+import { Modal, ModalProps } from 'components/Modal';
+
+import styled from 'styled-components';
+
+type TransasctionModalProps = Pick<ModalProps, 'isOpen' | 'onDismiss' | 'onCancel'>;
+
+export const Container = styled.div`
+  display: flex;
+
+  flex-direction: column;
+
+  align-items: center;
+
+  padding: 24px;
+
+  background: #ffffff;
+
+  border: 1px solid #000000;
+
+  box-sizing: border-box;
+
+  box-shadow: 0px 0px 16px rgba(0, 0, 0, 0.25);
+
+  border-radius: 10px;
+`;
+
+export const ContainerText = styled.div`
+  font-family: Inter;
+
+  font-style: normal;
+
+  font-weight: ${props => props.fontWeight || 600};
+
+  font-size: ${props => props.fontSize || '16px'};
+  line-height: ${props => props.lineHeight || '24px'};
+  display: flex;
+
+  align-items: center;
+
+  text-align: center;
+
+  color: ${props => props.color || '#000000'};
+`;
+
+const TransactionSubmittedContent: React.ReactElement = (
+  <div>
+    <div>Transaction Submitted</div>
+
+    <div>View on Block Explorer</div>
+  </div>
+);
+
+const TransactionRejectedContent: React.ReactElement = (
+  <div>
+    <div>Transaction Rejected</div>
+
+    <div>View on Block Explorer</div>
+  </div>
+);
+
+//@TODO: make dynamic
+
+const WaitingForTransactionContent: React.ReactElement = (
+  <Container>
+    <PendingCircle height="86px" width="86px" color="black" />
+
+    <div style={{ marginBottom: '16px', marginTop: "16px" }}>
```

Comment:
no inline styles pls...

Reply:
I thought it would be excessive to create a component just to set the margins. I will create a unique component for this.

Reply:
yes, although is only used once, is good idea to encapsulate it in a styled component. You can extend other component with styles too if you need to add that margin but in this case is a container for more than 1 element.